### PR TITLE
IoT care pet example

### PR DESCRIPTION
**Pull request by sitano**
Care Pet example
===

This is an example project that demonstrates a generic
IoT use case, using ScyllaDB Go driver infrastructure.

The application allows tracking of pets health indicators
and consist of 3 parts:

- migrate (`/cmd/migrate`) - bootstraps database schema
- collar sensor (`/cmd/sensor`) - accumulates pet health data and publish it
- web app (`/cmd/server`) - REST API for tracking pets state

### CODE REVIEW

File: iot-care-pet-example/go/README.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+Care Pet example
+===
+
+This is an example project that demonstrates basic usage
+patterns of ScyllaDB Go driver infrastructure.
```
Comment (tzach) [inline iot-care-pet-example/go/README.md:5]:
```suggestion
This is an example project that demonstrates a generic IoT use case, using ScyllaDB Go driver infrastructure.
```

File: iot-care-pet-example/go/README.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+- migrate (`/cmd/migrate`) - bootstrap database schema
+- collar sensor (`/cmd/sensor`) - accumulates pet health data and publish it
+- web app (`/cmd/server`) - REST API for tracking pets state
+
+Quick Start
+---
+
+Compose will spin 3 ScyllaDB nodes pinned to the first core.
+
+    $ docker-compose up
```
Comment (tzach) [inline iot-care-pet-example/go/README.md:19]: AFAIU, this command never exit. If this is the case the next command should be in a different terminal
(or run the first in the background)

File: iot-care-pet-example/go/README.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+The application allows tracking of pets health indicators
+and consist of 3 parts:
+
+- migrate (`/cmd/migrate`) - bootstrap database schema
+- collar sensor (`/cmd/sensor`) - accumulates pet health data and publish it
+- web app (`/cmd/server`) - REST API for tracking pets state
+
+Quick Start
+---
```
Comment (tzach) [inline iot-care-pet-example/go/README.md:15]: prerequisites:
- Docker and Docker Compose
- Go

File: iot-care-pet-example/go/README.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+Compose will spin 3 ScyllaDB nodes pinned to the first core.
+
+    $ docker-compose up
+    $ docker exec -it carepet-scylla1 cqlsh
+
+To initialize database execute:
+
+    $ go build ./cmd/migrate
+    $ ./migrate --hosts carepet-scylla1
```
Comment (tzach) [inline iot-care-pet-example/go/README.md:25]: I got the following error
```
./migrate --hosts carepet-scylla1
2020/08/06 10:04:56 Bootstrap database...
2020/08/06 10:04:57 gocql: dns error: lookup carepet-scylla1 on 192.168.0.1:53: no such host
2020/08/06 10:04:57 session:  gocql: unable to create session: failed to resolve any of the provided hostnames
```

Altho this did work
```
./migrate --hosts carepet-scylla1
2020/08/06 10:04:56 Bootstrap database...
2020/08/06 10:04:57 gocql: dns error: lookup carepet-scylla1 on 192.168.0.1:53: no such host
2020/08/06 10:04:57 session:  gocql: unable to create session: failed to resolve any of the provided hostnames
```

File: iot-care-pet-example/go/README.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+To start pet collar simulation:
+
+    $ go build ./cmd/sensor
+    $ ./sensor --hosts carepet-scylla1
```
Comment (tzach) [inline iot-care-pet-example/go/README.md:30]: same issue 
```
./sensor --hosts carepet-scylla1
2020/08/06 10:06:59 Welcome to the Pet collar simulator
2020/08/06 10:06:59 gocql: dns error: lookup carepet-scylla1 on 192.168.0.1:53: no such host
2020/08/06 10:06:59 session:  gocql: unable to create session: failed to resolve any of the provided hostnames

```

File: iot-care-pet-example/go/README.md (hunk 1 above)
Comment (sitano) [reply iot-care-pet-example/go/README.md]: Accepted

File: iot-care-pet-example/go/README.md (hunk 3 above)
Comment (sitano) [reply iot-care-pet-example/go/README.md:15]: Accepted

File: iot-care-pet-example/go/README.md (hunk 2 above)
Comment (sitano) [reply iot-care-pet-example/go/README.md]: Changed to detach with `-d`

File: iot-care-pet-example/go/README.md (hunk 3 above)
Comment (sitano) [reply iot-care-pet-example/go/README.md:15]: Accepted

File: iot-care-pet-example/go/README.md (hunk 4 above)
Comment (sitano) [reply iot-care-pet-example/go/README.md]: Rewrote into:

```
To initialize database execute:

    $ go build ./cmd/migrate
    $ NODE1=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' carepet-scylla1)
    $ ./migrate --hosts $NODE1
    
    expected output:
    
    2020/08/06 16:43:01 Bootstrap database...
    2020/08/06 16:43:13 Keyspace metadata = {Name:carepet DurableWrites:true StrategyClass:org.apache.cassandra.locator.NetworkTopologyStrategy StrategyOptions:map[datacenter1:3] Tables:map[gocqlx_migrate:0xc00016ca80 measurement:0xc00016cbb0 owner:0xc00016cce0 pet:0xc00016ce10 sensor:0xc00016cf40 sensor_avg:0xc00016d070] Functions:map[] Aggregates:map[] Types:map[] Indexes:map[] Views:map[]}
```

File: iot-care-pet-example/go/README.md (hunk 5 above)
Comment (sitano) [reply iot-care-pet-example/go/README.md]: rewrote into:

```
To start pet collar simulation:

    $ go build ./cmd/sensor
    $ NODE1=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' carepet-scylla1)
    $ ./sensor --hosts $NODE1
    
    expected output:
    
    2020/08/06 16:44:33 Welcome to the Pet collar simulator
    2020/08/06 16:44:33 New owner # 9b20764b-f947-45bb-a020-bf6d02cc2224
    2020/08/06 16:44:33 New pet # f3a836c7-ec64-44c3-b66f-0abe9ad2befd
    2020/08/06 16:44:33 sensor # 48212af8-afff-43ea-9240-c0e5458d82c1 type L new measure 51.360596 ts 2020-08-06T16:44:33+02:00
    2020/08/06 16:44:33 sensor # 2ff06ffb-ecad-4c55-be78-0a3d413231d9 type R new measure 36 ts 2020-08-06T16:44:33+02:00
    2020/08/06 16:44:33 sensor # 821588e0-840d-48c6-b9c9-7d1045e0f38c type L new measure 26.380281 ts 2020-08-06T16:44:33+02:00
    ...
    
Write down the pet Owner ID and pet ID (ID is something after the `#` sign without trailing spaces).
```

**Comment by tzach** [comment]
Closing. We now have the care pet project https://github.com/scylladb/care-pet

**Pull request closed without merging**